store/db.go

```diff
@@ -29,9 +30,36 @@ const (
 	DbFilename = "ql.db"
 )
 
+// dbLock is used to guarantee both thread-safety and process-safety
+// for db access.
+type dbLock struct {
+	// Mutex here is not necessary in fact, this is just to
```

Comment:
Nits:
a) list the mutex last in the struct (by convention it's listed before things it protects, and it doesn't really protect `lock.FileLock`)
b) change comment:
```
// Although the FileLock already ensures thread safety, the Go runtime is unaware 
// of this, and so Mutex is necessary to satisfy the Go race detector
```